SignalMessaging/utils/OWSWindowManager.m

```diff
@@ -3,29 +3,44 @@
 //
 
 #import "OWSWindowManager.h"
-#import "Signal-Swift.h"
-#import <SignalMessaging/UIColor+OWS.h>
-#import <SignalMessaging/UIFont+OWS.h>
-#import <SignalMessaging/UIView+OWS.h>
+#import "UIColor+OWS.h"
+#import "UIFont+OWS.h"
+#import "UIView+OWS.h"
+#import <SignalMessaging/SignalMessaging-Swift.h>
 
 NS_ASSUME_NONNULL_BEGIN
 
+NSString *const OWSWindowManagerCallDidChangeNotification = @"OWSWindowManagerCallDidChangeNotification";
+
+
+const CGFloat OWSWindowManagerCallScreenHeight(void)
+{
+    if ([UIDevice currentDevice].isIPhoneX) {
+        // On an iPhoneX, the system return-to-call banner has been replaced by a much subtler green
+        // circle behind the system clock. Instead, we mimic the old system call banner as on older devices,
+        // but it has to be taller to fit beneath the notch.
+        return 64;
+    } else {
+        return CurrentAppContext().statusBarHeight + 20;
```

Comment:
I feel somewhat nervous about all of the places that we are special-casing iPhone X or baking in iOS size constants.  

I wonder if we should leave in a new kind of code marker (like TODO), maybe IOS_DEVICE_CONSTANT or something that we can use to remind ourselves of every constant we should check whenever there's a new major iOS update or new device model.

Reply:
Unfortunately, I'm not sure that there is a constant that gets us what we want. This kind of layout is so far against the grain of what iOS asks us for, though I tried, I couldn't figure out computable constants for some of them.

I have annotated them with the IOS_DEVICE_CONSTANT comment tag, and also put a note in the sentinel in the UIDevice+featureSupport, so that if we get a new device size and hit that assert we'll have some direction about the things to fix.